src/SLCore/Analysis/SLCoreRuleSettings.cs

```diff
@@ -0,0 +1,81 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.ComponentModel.Composition;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.UserRuleSettings;
+using SonarLint.VisualStudio.SLCore.Core;
+using SonarLint.VisualStudio.SLCore.Service.Rules;
+using SonarLint.VisualStudio.SLCore.Service.Rules.Models;
+
+namespace SonarLint.VisualStudio.SLCore.Analysis;
+
+public interface ISLCoreRuleSettings
+{
+    Dictionary<string, StandaloneRuleConfigDto> RulesSettings { get; }
```

Comment:
Convert to Get.. method as this property does some calculation inside and does not cache the value

Reply:
To me it's just a bit of a misuse of the property. It's a fetch + convert, which is a complicated operation, and that should be a method. https://stackoverflow.com/questions/1374273/when-to-use-properties-instead-of-functions#:~:text=Use%20a%20property%20when%20the,ToString.